components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -156,12 +137,27 @@ func (g *gpsrtk) receiveAndWriteCorrectionData() {
 	defer g.activeBackgroundWorkers.Done()
 	defer g.closeCorrectionWriter()
 
-	err := g.connectToNTRIP()
+	select {
+	case <-g.cancelCtx.Done():
+		g.err.Set(errors.New("context canceled"))
+		return
+	default:
+	}
+
+	err := g.connectAndParseSourceTable()
 	if err != nil {
 		g.err.Set(err)
-		g.logger.Error("unable to connect to NTRIP stream! Giving up on RTK messages")
+		g.logger.Error("unable to parse source table! Giving up on RTK messages")
 		return
 	}
+
+	g.reader, err = g.getStream()
+	if err != nil {
+		g.err.Set(err)
+		g.logger.Error("unable to get NTRIP stream! Giving up on RTK messages")
```

Comment:
Same here, We  should add the error that getStream gives. 
("Failed to get NTRIP stream ", err)

Reply:
I meant for the "giving up" part to indicate that the entire background goroutine is stopping permanently and if you ever want NTRIP updates in the future you need to rebuild the component. I still think it's important to say that, but I guess I messed up the phrasing. How should I have said it instead, do you think?

Reply:
how about "Failed to retrieve NTRIP stream! Aborting" ???